### ISSUE TITLE
Naming clash for new cmp standard package

### ISSUE THREAD

**Issue by Carrotman42** [google/go-cmp#354]
I am not sure where the best place to have this discussion is, but I wanted to get @dsnet's opinion on this so this seemed like a reasonable place to start.

This is an extremely popular package and is imported into many Go tests across the world. Unfortunately, the new [`cmp` package](https://pkg.go.dev/cmp) added within the last year has the same name as this package. I can see one possibly likely scenario where I would want to import both packages: in a unit test which wanted to use cmp.Diff along with a cmpopts.SortSlice which used cmp.Less[sometype]. There may be other situations as well.

The general solution to two packages having the same name is to rename one of them, but it isn't clear exactly which should be renamed: while it seems like a package in the stdlib is more central (leading to a preference to rename the third-party go-cmp), since go-cmp predates std/cmp by many years it will be a burden to retrain people with this new rename rule.

Does this package (or its maintainers) have an opinion about how to handle this naming clash?

PS: One out-of-the-box idea I had was to add forwarding code in go-cmp to all things in std/cmp, but I think that would cause people to import the heavier-weight go-cmp package into non-test code by accident so I am not certain that's a great idea.

**Comment by dsnet** [comment]
The name collision is unfortunate and was brought up during the proposal for the stdlib "cmp" package. The concern was dismissed as not being a problem: https://github.com/golang/go/issues/59488#issuecomment-1554911781

Adding "go-cmp/cmp" to the stdlib has been proposed before, but rejected: https://github.com/golang/go/issues/45200

I personally still find the name collision unfortunate, but don't see the benefit of just renaming "go-cmp/cmp" worth the cost in churn. If we were to do a v2 of the package or entirely move it elsewhere (e.g., the stdlib), there there might be enough benefits to warrant a rename as well.

**Comment by neild** [comment]
The standard library `cmp` is obviously not being renamed.

We could rename `github.com/google/go-cmp/cmp`. But that's a lot of churn and confusion, for a questionable benefit.

I think the best thing to do for now is wait and see. Perhaps this isn't an issue at all, or is a minor one. We'll have more information in a year or two.

**Comment by mxey** [comment]
I agree on the churn. It would be nice to have an agreed upon suggested alias for the import, so people would clearly recognise it in other people's Go code.

**Comment by mrwormhole** [comment]
aliasing as gcmp or gocmp is the only solution forward, it is not that std cmp is extremely useful for every test file, it is very small addition to std lib so chances are slim that you will use both at the same time

**Comment by cespare** [comment]
By the way, in my experience, the issue was initially much more minor because very little test code used `cmp`. The frog has been getting boiled through the addition of subsequent APIs like `slices.SortFunc`. As comparison-based ordering functions become the norm, we use `cmp.Compare` more and more in all kinds of code, including test code.
